python/ray/tune/trainable/function_trainable.py

```diff
@@ -58,6 +59,7 @@ def setup(self, config):
                 logdir=self._storage.trial_driver_staging_path,
                 driver_ip=None,
                 experiment_name=self._storage.experiment_dir_name,
+                run_id=uuid.uuid4().hex,
```

Comment:
Create a unique ID that differentiate each run.

- Cannot use trial id because of `trainer.restore` will reuse trial id.
- Cannot use job id because there could be multiple train runs in one job.
- Cannot use trial id + job id because one can restore a run multiple times in one job.